smartapps/smartthings/hue-connect.src/hue-connect.groovy

```diff
@@ -879,36 +877,40 @@ private handleCommandResponse(body) {
 	// scan entire response before sending events to make sure they are always in the same order
 	def updates = [:]
 
-                body.each { payload ->
-                    log.debug $payload
+	body.each { payload ->
+		log.debug $payload
 
 		if (payload?.success) {
-                        def childDeviceNetworkId = app.id + "/"
-                        def eventType
+			def childDeviceNetworkId = app.id + "/"
+			def eventType
 			payload.success.each { k, v ->
 				def data = k.split("/")
 				if (data.length == 5) {
 					childDeviceNetworkId = app.id + "/" + k.split("/")[2]
 					if (!updates[childDeviceNetworkId])
 						updates[childDeviceNetworkId] = [:]
-                            eventType = k.split("/")[4]
+					eventType = k.split("/")[4]
 					updates[childDeviceNetworkId]."$eventType" = v
 				}
 			}
 		} else if (payload.error) {
 			log.warn "Error returned from Hue bridge error = ${body?.error}"
-                            }
-                        }
+		}
+	}
 
 	// send events for each update found above (order of events should be same as handlePoll())
 	updates.each { childDeviceNetworkId, params ->
 		def device = getChildDevice(childDeviceNetworkId)
-		sendBasicEvents(device, "on", params.on)
-		sendBasicEvents(device, "bri", params.bri)
-		sendColorEvents(device, params.xy, params.hue, params.sat, params.ct)
-                    }
+		def id = getId(device)
+		// If device is offline, then don't send events which will update device watch
+		if (isOnline(id)) {
+			sendBasicEvents(device, "on", params.on)
```

Comment:
So would these events be sent on some kind of schedule as long as a device remain online?